src/AspNet.Security.OAuth.Automatic/AutomaticAuthenticationHandler.cs

```diff
@@ -0,0 +1,78 @@
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+
+namespace AspNet.Security.OAuth.Automatic {
+    internal class AutomaticAuthenticationHandler : OAuthHandler<AutomaticAuthenticationOptions> {
+        public AutomaticAuthenticationHandler([NotNull] HttpClient httpClient)
+            : base(httpClient) {}
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens) {
+            var request = new HttpRequestMessage(HttpMethod.Get, Options.UserInformationEndpoint);
+            request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", tokens.AccessToken);
+
+            var response = await Backchannel.SendAsync(request, Context.RequestAborted);
+            if (!response.IsSuccessStatusCode) {
+                Logger.LogError("An error occurred when retrieving the user profile: the remote server " +
+                                "returned a {Status} response with the following payload: {Headers} {Body}.",
+                    /* Status: */ response.StatusCode,
+                    /* Headers: */ response.Headers.ToString(),
+                    /* Body: */ await response.Content.ReadAsStringAsync());
+
+                throw new HttpRequestException("An error occurred when retrieving the user profile.");
+            }
+
+            var payload = JObject.Parse(await response.Content.ReadAsStringAsync());
+
+            identity.AddOptionalClaim(ClaimTypes.NameIdentifier, AutomaticHelper.GetIdentifier(payload),
+                    Options.ClaimsIssuer)
+                .AddOptionalClaim(ClaimTypes.GivenName, AutomaticHelper.GetGivenName(payload), Options.ClaimsIssuer)
+                .AddOptionalClaim(ClaimTypes.Surname, AutomaticHelper.GetFamilyName(payload), Options.ClaimsIssuer)
+                .AddOptionalClaim(ClaimTypes.Name, AutomaticHelper.GetLogin(payload), Options.ClaimsIssuer)
+                .AddOptionalClaim(ClaimTypes.Email, AutomaticHelper.GetEmail(payload), Options.ClaimsIssuer)
+                .AddOptionalClaim(ClaimTypes.Uri, AutomaticHelper.GetUrl(payload), Options.ClaimsIssuer);
+
+            var principal = new ClaimsPrincipal(identity);
+            var ticket = new AuthenticationTicket(principal, properties, Options.AuthenticationScheme);
+
+            var context = new OAuthCreatingTicketContext(ticket, Context, Options, Backchannel, tokens, payload);
+            await Options.Events.CreatingTicket(context);
+
+            return context.Ticket;
+        }
+
+        protected override string BuildChallengeUrl(AuthenticationProperties properties, string redirectUri) {
+            var scopes = FormatScope();
+
+            var queryStrings = new Dictionary<string, string> {
```

Comment:
> No prob, is the next release still going to be beta2 now that Core 1.0 is final?

Yep. No idea when it will be "officially" released on NuGet.org, tho.